Proxies/Numbers.cs

```diff
@@ -29,7 +29,7 @@ public abstract class IntegerProxy {
         }
 
         [JSIsPure]
-        [JSReplacement("($this === $rhs)")]
+        [JSReplacement("(($this).valueOf() === ($rhs).valueOf())")]
```

Comment:
Really we should split it into to separate.
T.Equals(T) should be left as $this === $rhs
T.Equal(object) should be $this === $rhs.value().
Hm, really there may be null or even undefined. Than I think we don't need override in special way `T.Equal(object)` for numeric type. It will work slow, but it should not be called often.